packages/components/Notification/src/Notification.tsx

```diff
@@ -41,9 +41,15 @@ export const Notification = compose<NotificationType>({
   useRender: (userProps: NotificationProps, useSlots: UseSlots<NotificationType>) => {
     const Slots = useSlots(userProps, (layer) => notificationLookup(layer, userProps));
     const isBar = ['primaryOutlineBar', 'primaryBar', 'neutralBar'].includes(userProps.variant);
+    const width = useWindowDimensions().width / 2;
```

Comment:
I'm going to guess no since slotProps is an object and not a hook itself.

Reply:
But you can try - we have tools that will warn us if a hook is not allowed in a block of code.

Reply:
Unfortunately not ://. The hook wouldn't be at top level anymore

Reply:
Thanks. My followup question.. now that we're almost done, is there any code worthy of being combined into a `useNotification` hook? Looking back, I think the issue we had is generally the hook creates the `{ props, state }` for the whole component, which you might put on the root slot. For this component, we did a lot of calculating props for each slot (see lines 63-64). I wasn't quite sure if that fits in the `use<Component>` hook paradigm? Do you have any thoughts @rurikoaraki ?  I was actually thinking I'll ask the wider team later today because I wasn't sure when to make those types of hooks.

Reply:
You can change the hook to return a different type which includes slot information. I think checkbox does that.

The main reason to mess with props outside of useSlot is when you need children in order to determine prop information. Otherwise it's useful to keep it in use<Component> so that 1. it's one place to look for all the meaningful work and 2. if a client wants to rewrite the render function to say, add an additional slot or something, it's less for them to rewrite if stuff is in use<Component> and they can just take it and add to it.

Reply:
Thanks. I think we'll go ahead with this PR and have a `useNotification` refactor as a followup if it works out.